src/sentry/sentry_metrics/multiprocess.py

```diff
@@ -536,12 +537,31 @@ def __init__(
         self.__producer_topic = settings.KAFKA_TOPICS[settings.KAFKA_SNUBA_METRICS].get(
             "topic", "snuba-metrics"
         )
+        self.__commit_function = commit_function
 
         self.__closed = False
         self.__metrics = get_metrics()
+        self.__produced_message_offsets = []
+        self.__callbacks = 0
 
     def poll(self) -> None:
-        pass
+        if self.__callbacks and self.__produced_message_offsets:
```

Comment:
Please add here the same metric we had in the collector. We need to be able to compare the results apples to apples.
This one 
`with self.__metrics.timer("simple_produce_step.join_duration"):` 

Specifically, we need two metrics since you cannot record a metric each time we call poll (which means at every cycle):
- Log a timing metric that covers the poll method (sampled, like log it every 100 messages or so)
- accumulate the time you spend on poll and once in a while, log `simple_produce_step.join_duration` with that entire duration. We will use it to compare the sum of time spent overall in poll vs transform. We will not use that for avg/p99 obviously, for that we will use the sampled metric.